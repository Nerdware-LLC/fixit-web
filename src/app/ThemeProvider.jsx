import React from "react";
import { ThemeProvider as MTP, createTheme } from "@mui/material/styles";
import CssBaseline from "@mui/material/CssBaseline";
import { any } from "../types";

export const ThemeProvider = ({ children }) => {
  const theme = createTheme(THEME);

  return (
    <MTP theme={theme}>
      <CssBaseline />
      {children}
    </MTP>
  );
};

const THEME = {
  description: "dark",
  palette: {
    mode: "dark",
    primary: {
      light: "#f89a35",
      main: "#f78103", // Fixit orange
      dark: "#ac5a02"
    },
    secondary: {
      light: "#5381ff",
      main: "#2962ff", // blue
      dark: "#1c44b2"
    },
    success: {
      light: "#33d373",
      main: "#00c851",
      dark: "#008c38"
    },
    error: {
      main: "rgba(255, 115, 115, 0.87)"
    },
    info: {
      main: "#ff80ab" // AKA NOTIFICATION
    },
    text: {
      primary: "rgba(255, 255, 255, 0.87)",
      disabled: "rgba(255, 255, 255, 0.38)",
      hint: "rgba(255, 255, 255, 0.54)", // AKA PLACEHOLDER
      labels: "#f7a44a" // desaturated Fixit orange (CUSTOM KEY)
    },
    background: {
      paper: "#353535", // AKA SURFACE
      default: "#121212",
      backdrop: "rgba(0, 0, 0, 0.5)" // (CUSTOM KEY)
    }
  }
};

ThemeProvider.propTypes = {
  children: any.isRequired
};
