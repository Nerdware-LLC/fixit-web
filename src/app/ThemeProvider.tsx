import React from "react";
import MuiThemeProvider from "@mui/material/styles/ThemeProvider";
import CssBaseline from "@mui/material/CssBaseline";
import { createTheme, type PaletteOptions } from "@mui/material/styles";
import { themeStore } from ".";
import { any } from "../types";

export const ThemeProvider = ({ children }: { children: React.ReactNode }) => {
  const theme = themeStore.useSubToStore();

  return (
    <MuiThemeProvider theme={THEMES[theme]}>
      <CssBaseline />
      {children}
    </MuiThemeProvider>
  );
};

/**
 * `ThemeProvider` merges this theme object with the @mui/material
 * default theme:
 *
 * https://mui.com/material-ui/customization/default-theme/
 */
const THEMES = Object.fromEntries(
  ["DARK", "LIGHT"].map((themeModeName) => [
    themeModeName,
    createTheme({
      palette: {
        mode: themeModeName.toLowerCase() as PaletteOptions["mode"],
        primary: {
          light: "#f89a35",
          main: "#f78103", // Fixit orange
          dark: "#ac5a02"
        },
        secondary: {
          light: themeModeName === "DARK" ? "#00FFDC" : "#625AFA",
          main: themeModeName === "DARK" ? "#00FFDC" : "#625AFA",
          dark: themeModeName === "DARK" ? "#00FFDC" : "#625AFA"
        },
        success: {
          light: "#33d373",
          main: "#00c851",
          dark: "#008c38"
        },
        error: {
          main: "rgba(255, 115, 115, 0.87)"
        },
        info: {
          main: "#5796D6" // AKA NOTIFICATION
        },
        text: {
          primary: themeModeName === "DARK" ? "rgba(255, 255, 255, 0.87)" : "rgba(0, 0, 0, 0.87)",
          secondary: themeModeName === "DARK" ? "#00FFDC" : "#625AFA",
          disabled: "rgba(255, 255, 255, 0.38)",
          hint: "rgba(255, 255, 255, 0.54)", // AKA PLACEHOLDER
          labels: "#f7a44a" // desaturated Fixit orange (CUSTOM KEY)
        } as any, // <-- Custom theme properties are fine.
        background: {
          paper: themeModeName === "DARK" ? "#353535" : "#cccccc", // AKA SURFACE
          default: themeModeName === "DARK" ? "#121212" : "#f6f6f6",
          backdrop: "rgba(0, 0, 0, 0.5)" // (CUSTOM KEY)
        } as any // <-- Custom theme properties are fine.
      },
      typography: {
        fontFamily: '"Roboto", sans-serif'
      }
    })
  ])
);

ThemeProvider.propTypes = {
  children: any.isRequired
};
