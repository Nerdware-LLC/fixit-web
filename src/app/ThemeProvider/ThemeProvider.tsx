import CssBaseline from "@mui/material/CssBaseline";
import MuiThemeProvider from "@mui/material/styles/ThemeProvider";
import { usePageLayoutContext } from "@app/PageLayoutContext/usePageLayoutContext";
import { themeStore } from "@app/apolloCache/reactiveVars/themeStore";
import { THEMES } from "./themes";
import type { Theme } from "@mui/material/styles";

export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const themeName = themeStore.useSubToStore();
  const { isMobileUserAgent, isMobilePageLayout } = usePageLayoutContext();

  const theme: Theme = {
    ...THEMES[themeName || "DARK"], // A fallback themeName for added safety (will crash if not set)
    variables: {
      isMobileUserAgent,
      isMobilePageLayout,
    },
  };

  return (
    <MuiThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </MuiThemeProvider>
  );
};

export type ThemeProviderProps = { children: React.ReactNode };
