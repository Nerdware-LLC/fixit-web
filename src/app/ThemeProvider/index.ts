export { ThemeProvider } from "./ThemeProvider";
export * from "./themes";
