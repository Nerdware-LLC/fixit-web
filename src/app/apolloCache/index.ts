export { apolloCache } from "./apolloCache";
export * from "./reactiveVars";
