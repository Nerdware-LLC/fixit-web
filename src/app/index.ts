export { App } from "./App";

// Also export apollo-cache reactive vars:
export * from "./apolloCache/reactiveVars";
