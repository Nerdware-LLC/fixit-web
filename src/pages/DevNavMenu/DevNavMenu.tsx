import { Link } from "react-router-dom";
import styled from "@emotion/styled";
import { Title } from "../../components";

export const DevNavMenu = () => (
  <StyledDevNavMenuLayout>
    <Title>Dev Nav Menu - Links to All Pages</Title>
    <Link to="/">Landing Page</Link>
    <Link to="/login">Login Page</Link>
    <Link to="/register">Register Page</Link>
    <Link to="/products">Products Page</Link>
    <Link to="/checkout">
      {/* NOTE: this will error, bc query params "token", "sub", and "promoCode" are not set */}
      Checkout Page
    </Link>
    <Link to="/home">Home Page</Link>
    <Link to="/home/connect/return">Stripe Connect Onboarding Bridge (RETURN)</Link>
    <Link to="/home/connect/return">Stripe Connect Onboarding Bridge (REFRESH)</Link>
    <Link to="/home/customer-portal">Stripe Customer Portal Bridge</Link>
    <Link to="/page-not-found">"Page Not Found" Page</Link>
  </StyledDevNavMenuLayout>
);

const StyledDevNavMenuLayout = styled.div`
  box-sizing: border-box;
  height: 100%;
  text-align: left;
  display: flex;
  flex-direction: column;
  padding: 30vh 15vw;
  align-items: center;
  justify-content: space-around;
`;
