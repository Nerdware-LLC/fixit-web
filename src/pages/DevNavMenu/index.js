export { DevNavMenu as default } from "./DevNavMenu";
// export as default to enable code-splitting via lazy import.
