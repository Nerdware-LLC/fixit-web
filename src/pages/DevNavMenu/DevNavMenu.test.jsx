import renderer from "react-test-renderer";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { DevNavMenu } from "./DevNavMenu";
import { ThemeProvider } from "../../app/ThemeProvider";

it("renders correctly", () => {
  expect(
    renderer
      .create(
        <ThemeProvider>
          <MemoryRouter initialEntries={["/"]}>
            <Routes>
              <Route index element={<DevNavMenu />} />
            </Routes>
          </MemoryRouter>
        </ThemeProvider>
      )
      .toJSON()
  ).toMatchSnapshot();
});
