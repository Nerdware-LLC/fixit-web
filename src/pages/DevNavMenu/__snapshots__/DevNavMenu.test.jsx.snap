// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-15qcvv9"
>
  <p
    className="css-mafmc5"
  >
    Dev Nav Menu - Links to All Pages
  </p>
  <a
    href="/checkout"
    onClick={[Function]}
  >
    Checkout Page
  </a>
  <a
    href="/connect/return"
    onClick={[Function]}
  >
    Stripe Connect Onboarding Bridge (RETURN)
  </a>
  <a
    href="/connect/return"
    onClick={[Function]}
  >
    Stripe Connect Onboarding Bridge (REFRESH)
  </a>
  <a
    href="/customer-portal"
    onClick={[Function]}
  >
    Stripe Customer Portal Bridge
  </a>
  <a
    href="/page-not-found"
    onClick={[Function]}
  >
    "Page Not Found" Page
  </a>
</div>
`;
